### ISSUE TITLE
v3.11.0-b BETA - Approximate address allowed for an in-person meeting that isn't 'Location temporarily closed'

### ISSUE THREAD

**Issue by courage2change** [code4recovery/12-step-meeting-list#388]
**Is your feature request related to a problem? Please describe.**
I have an issue with the new way of categorising meetings as online/in-person/location temporarily closed just by the location and Zoom/phone fields.

**Describe the solution you'd like**
Previously, it seemed to me to be possible to add an in-person meeting with an approximate address.  This is a really important feature for certain in-person meetings which are not temporarily closed, but for various reasons don't want to publicise their exact location online and want the newcomer to contact them for the address.  
With the 3.11 beta using the location field and Zoom/phone fields solely to determine whether the meeting is online or not, and not allowing in person meetings to have an approximate address, this restricts the flexibility of the plugin.
Not sure how this would work for hybrid meetings, as these are determined by the precision of the location field too.
It would be great to return to having a tickbox for 'Location temporarily closed', and allowing in person and hybrid meetings to have an approximate physical address - maybe another tickbox - 'This meeting's physical location is active but does not want to publicise its location'?

**Describe alternatives you've considered**
Would have to find an alternative if this functionality isn't available.

**Additional context**
Thanks for all your service, love the new UI.

**Comment by brianw-area24** [comment]
Thank you for testing v3.11.0-b of the Plugin. Having a few people that can run the new version, and provide feedback before the full release is very helpful to us!

We've had a couple discussions while working on this version, and the new TSML-UI, about in-person meetings with approximate addresses. We decided to treat these as temporarily closed for a couple of reasons.

1. It's my understanding that this is how the Meeting Guide App treats these meetings, and we want to be consistent with how they are handling these meetings
2. You can't attend an in-person meeting without a specific address, and we see the situation you describe as an edge case. This requires detailed information in the notes field(s), and we have no way of ensure that's happening
3. The addition of approximate addresses is a pretty recent feature of the plugin, after we started supporting Online meetings

If you know how your site treated these meetings before COVID hit, and we started allowing approximate addresses at all, I'd be interested in hearing about how they were handled.

Again, thank you so much for helping us test this new version! I hope it's been a good experience except for the couple things you mentioned in the issues queue here.

Brian W.

**Comment by courage2change** [comment]
Thanks very much for that Brian.  

We haven't started using the plugin yet as the site is being developed, but on our current site, most meetings have approximate addresses, even before COVID hit - there are lots of sensitivities which mean a meeting might not want to publicise the full address of a meeting.  I realise approximate addresses for in-person meetings are an edge case for most AA meetings and the Meeting Guide app, but it will be a must-have for some fellowships or types of meeting.  

Could you see an easy way around this so I can get this functionality, or do you think it's probably a case of looking for another solution?

**Comment by tech2serve** [comment]
Hi @courage2change—

Let me echo @brianw-area24’s comments thanking you for testing the code. Most important to us.

It is my understanding that MG will drop meetings that do not have online conference/phone information and a non-specific address (i.e., an approximate address). So allowing these to be published essentially provided a false sense for the admin that those meetings would be included in Meeting Guide. We consider this a bad situation :-)

We also recognize the situation you’re describing; and yes it is an edge case, at least for AA. What fellowship are you maintaining meetings for? Code for Recovery can discuss more possible approaches, but as Brian said, we’ve got to maintain a good data feed for Meeting Guide.

If the situation you’ve described is common for your fellowship/site, I can think of a couple workarounds or solutions.

1. Use your CSO or some obvious non-meeting place as the address for those meetings. This then requires notes that tell users there isn’t actually a meeting there, but that they need to call a number to get the actual address.
2. In theory, you could put your meeting list behind a sign-in wall. That is outside the scope of any support we would have the time to provide.
3. You could create a child theme for your site, and use the `functions.php` to override the behavior of handling approximate meetings. You’ll need to use your favorite search engine to learn how to do this.
4. You may want to fork the code and develop a version that handles the meetings the way you wish. This is a bit more involved in the sense that you’d likely want to maintain synchronization with our development.

As I’ve said, we can discuss this further as a group, but I would like to hear back from you regarding your fellowship.

**Comment by joshreisner** [comment]
> It is my understanding that MG will drop meetings that do not have online conference/phone information and a non-specific address (i.e., an approximate address). 

This is true. I handled app support for many sites that had "roving" meetings or meetings at members' private homes, and didn't want to publicize the meetings for whatever reason. The other common support request along these lines was how to handle meetings that don't meet on a weekly cadence.) We recommended those sites create a separate page on their site to handle these special meetings, since they are an edge-case. 

I would recommend against option 1 above. I think putting a false address, such as the intergroup central office, is bound to cause confusion eventually, even with contrary signals in notes or other fields such as the meeting or location name.
